.github/workflows/build.yml

```diff
@@ -0,0 +1,151 @@
+name: Builld with IDF
+
+on:
+  - push
+
+jobs:
+  build_idf_master:
+    name: build-with-IDF-master
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout esp-protocols
+        uses: actions/checkout@master
+        with:
+          path: esp-protocols
+      - name: Checkout IDF
+        uses: actions/checkout@master
+        with:
+          repository:  espressif/esp-idf
+          path: esp-idf
+
+      - name: Build-examples-with-IDF
+        run: |
+          cd $GITHUB_WORKSPACE/esp-idf
+          git submodule update --init --recursive
+          ./install.sh
+          source export.sh
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/simple_cmux_client
+          idf.py build
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/ap_to_pppos
+          idf.py build
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/modem_console
+          idf.py build
+          cd $GITHUB_WORKSPACE/esp-protocols/components/esp_modem/examples/pppos_client
+          idf.py build
+
+  build_idf_v4_4:
+    name: build-with-IDF-4.4
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout esp-protocols
+        uses: actions/checkout@master
+        with:
+          path: esp-protocols
+      - name: Checkout IDF
+        uses: actions/checkout@master
+        with:
+          repository:  espressif/esp-idf
+          path: esp-idf
+          ref: release/v4.4
+
+      - name: Build-examples-with-IDF
+        run: |
+          cd $GITHUB_WORKSPACE/esp-idf
+          git submodule update --init --recursive
+          ./install.sh
```

Comment:
Instead of running `install.sh` in the build job, have you considered using versioned IDF Docker images?

See for example: https://github.com/igrr/libnmea-esp32/blob/ff80ce536dfa3f1be40fa2c8c1f37d542d958cdf/.github/workflows/build.yml#L8-L17

(https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/tools/idf-docker-image.html)

Reply:
Thanks for the suggestion! I've rewritten the yaml based on your example.